### ISSUE TITLE
Survey existing tools on documentation

### ISSUE THREAD

**Issue by choldgraf** [BIDS/docathon#4]

**Comment by choldgraf** [comment]
...and some projects that could be started on the technical side of things

**Comment by mrzv** [comment]
C++:
- [Doxygen](http://www.stack.nl/~dimitri/doxygen/)
- [BoostBook](http://www.boost.org/doc/libs/1_62_0/doc/html/boostbook.html) and [QuickBook](http://www.boost.org/doc/libs/1_62_0/doc/html/quickbook.html)
- [Breathe](https://breathe.readthedocs.io/en/latest/) for connecting Sphinx and Doxygen

**Comment by mrzv** [comment]
Should [StackOverflow Documentation](http://stackoverflow.com/documentation) somehow fit into this.

**Comment by choldgraf** [comment]
If we can find some specific things to reference, I say yes. But I think
it'll be most useful if it's a well-selected group of materials rather than
a fire hose of potential information.
--

**Comment by mrzv** [comment]
I was more contemplating it as a documentation tool, one that supports multiple languages. It's true that they only allow projects that are large enough. And it's not clear how well it's doing as a documentation tool. And the issue of having your documentation on StackOverflow is not a simple one. But I think it's worth keeping in mind, at least as an example of "documentation by example."

**Comment by choldgraf** [comment]
ahhh that's right, that could be interesting, at the very least worth
adding to the list of tools that we can point people towards.
--

**Comment by stefanv** [comment]
So far:

      - numpydoc
      - Napoleon
      - coverage tool of sorts?
      - sphinx extension: multiple languages
      
      From Matthias:

      - https://github.com/drdoctr/doctr
      - http://devdocs.io/
      - [[http://blog.readthedocs.com/announcing-pydoc-io/][pydoc.io]]  (exists since yesteraday), see [[https://github.com/rtfd/readthedocs.org/issues/1957#issuecomment-261647307][this issue]] as well
      - https://kapeli.com/dash
      - https://pypi.python.org/pypi/docrepr
      - https://github.com/sphinx-gallery/sphinx-gallery
      - https://github.com/michaelpacer/thebe-test

      - His new declarative Sphinx tool
        https://github.com/carreau/thoth

**Comment by Carreau** [comment]
Yes, thanks seen it. I'll try to setup a review for next meeting.

**Comment by choldgraf** [comment]
I updated the top-level comment w/ latest tools. Can you confirm that it's correct and modify as you like? We can keep updating this as more tools are given

**Comment by weiwzhang** [comment]
Some add-ons:

# Python:
- pydoctor https://launchpad.net/pydoctor
    - replace inactive Epydoc http://epydoc.sourceforge.net
- pdoc: https://github.com/BurntSushi/pdoc
- PyDoc https://docs.python.org/2/library/pydoc.html
- XIST: http://www.livinglogic.de/Python/xist/index.html
- HtmlGen (Python library) https://pypi.python.org/pypi/htmlgen/0.8
- Pandoc http://pandoc.org

# Others:
- ApiGen http://www.apigen.org (for PHP)
- appledoc: https://github.com/tomaz/appledoc (for Objective C)
- ClassDoc https://docs.oracle.com/javase/7/docs/jdk/api/javadoc/doclet/com/sun/javadoc/ClassDoc.html
- Javadoc: http://www.oracle.com/technetwork/articles/java/index-jsp-135444.html
- CPPDoc: http://www.cs.scranton.edu/~doc/cppdoc/  (for C++)

 [Cited from:]
    - pythonwiki https://wiki.python.org/moin/DocumentationTools
    - (list from)wikipedia: https://en.wikipedia.org/wiki/Comparison_of_documentation_generators
        - a list of 50+ documentation tools
        - tables of information about operational system support, language support, input/output format

**Comment by choldgraf** [comment]
all kinds of good stuff in there, TY!

**Comment by choldgraf** [comment]
Hey all - I just updated @Carreau 's edits to the resources page...shaping up nicely! I think at some point we may want to highlight some tools specifically...

https://bids.github.io/docathon/pages/resources.html

**Comment by meowklaski** [comment]
@choldgraf I noticed a typo for the `custom_inherit` description on the resources page. Seeing that I was a bit verbose earlier, here is a more succinct description:

>[custom_inherit](https://github.com/meowklaski/custom_inherit) is a Python package that provides convenient, light-weight tools for inheriting/merging docstrings in customizeable ways. This helps facilitate thorough and consistent documentation, since documentation need not be duplicated manually when it is being "inherited".

>It provides both a decorator and a metaclass for facilitating docstring inheritance. Numpy, Google, & reST docstring styles are supported out of the box.

**Comment by choldgraf** [comment]
cool - I've updated the language with your new wording, thanks for the correction!
